src/main/java/org/jenkinsci/extension_indexer/ExtensionPointListGenerator.java

```diff
@@ -95,37 +83,58 @@ public String getName() {
             return definition.extensionPoint;
         }
 
-        void formatAsConfluencePage(PrintWriter w) {
-            w.println("h2." + definition.extensionPoint);
-            w.println(getSynopsis(definition));
-            w.println(definition.confluenceDoc);
+        public String getShortName() {
+            if (getName().contains(".")) {
+                return getName().substring(getName().lastIndexOf(".") + 1);
+            }
+            return getName();
+        }
+
+        void formatAsAsciidoc(PrintWriter w) {
+            w.println();
+            w.println("## " + getShortName());
+            w.println("+" + definition.extensionPoint + "+");
+            w.println();
+            w.println(definition.documentation == null || formatJavadoc(definition.documentation).trim().length() == 0 ? "_This extension point has no Javadoc documentation._" : formatJavadoc(definition.documentation));
+            w.println();
+            w.println("**Implementations:**");
             w.println();
-            w.println("{expand:title=Implementations}");
             for (ExtensionSummary e : implementations) {
-                w.println("h3." + (e.implementation == null || e.implementation.equals("") ? "_Anonymous Class_" : e.implementation));
-                w.println(getSynopsis(e));
-                w.println(e.confluenceDoc == null ? "_This class has no Javadoc documentation._" : e.confluenceDoc);
+                w.println();
+                w.println((e.implementation == null || e.implementation.trim().length() == 0 ? "(Anonymous class)" : e.implementation) + " " + getSynopsis(e) + "::");
+                w.println((e.documentation == null || formatJavadoc(e.documentation).trim().length() == 0 ? "_This implementation has no Javadoc documentation._" : formatJavadoc(e.documentation)));
             }
             if (implementations.isEmpty())
-                w.println("(No known implementation)");
-            w.println("{expand}");
-            w.println("");
+                w.println("_(no known implementations)_");
+            w.println();
+        }
+
+        private String formatJavadoc(String javadoc) {
+            StringBuilder formatted = new StringBuilder();
+
+            for (String line : javadoc.split("\n")) {
```

Comment:
It is not going to work as you expect if the source file has `\r\n`. And we may have it for Parent POM 1.x